### PR TITLE
fix: not working together with the SSH FS extension

### LINKED ISSUE #536: Not working together with the SSH FS extension
hello,
i encounter an issue with Shellcheck :

It work like a charm on local folder, but when i use SSH FS extension to open a remote folder, i have an error who tell me shellcheck program was not found : 
image

This is my actual configuration : 

image

I have tried to set the executable path pointing on the last version of shellcheck, but with no success :/

Thanks for help ^^

### PR DESCRIPTION
**Pull request by timonwong**
Fix #536

### CODE REVIEW

**Comment by felipecrs** [comment]
I will test this as well, I was able to replicate the issue.

**Comment by felipecrs** [comment]
For me the same issue happened:

image

Did you test it as well?

**Comment by timonwong** [comment]
@felipecrs  Yes, however I encountered the error just during shellcheck process spawn.
Since we are using bundled shellcheck, so I guess ENOENT occurred when we are using an invalid working directory.

**Comment by timonwong** [comment]
image

**Comment by felipecrs** [comment]
Now it works for me too!

File: src/utils/tool-check.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export function tryPromptForUpdatingTool(version: semver.SemVer | null) {
 export async function getToolVersion(
   executable: string
 ): Promise<semver.SemVer | null> {
-  const { stdout } = execa.sync(executable, ["-V"], { timeout: 5000 });
+  const { stdout } = await execa(executable, ["-V"], { timeout: 5000 });
```
Comment (felipecrs) [inline src/utils/tool-check.ts:24]: Hey this fixed a bug, believe you or not:

https://github.com/vscode-shellcheck/vscode-shellcheck/pull/180

**Review by felipecrs — CHANGES_REQUESTED** [review]
I would suggest you split the changes about prettier formatting in a different commit if you really want to do it. It's a bit difficult to review this way..

File: src/utils/tool-check.ts (hunk 1 above)
Comment (felipecrs) [reply src/utils/tool-check.ts:24]: Here: https://github.com/vscode-shellcheck/vscode-shellcheck/pull/166

Comment (timonwong) [reply src/utils/tool-check.ts]: Ooops...

**Comment by timonwong** [comment]
Now I've restored th original prettier config. And add a another fix for file matchers

File: src/utils/path.ts [hunk 2]
```diff
@@ -1,27 +1,52 @@
+import path = require("path");
```
Comment (felipecrs) [inline src/utils/path.ts:1]:
```suggestion
import * as path from "path";
```

File: src/utils/path.ts [hunk 3]
```diff
@@ -1,27 +1,52 @@
...
       return fixDriveCasingInWindows(workspace.uri.fsPath);
     }
   }
 
   // fall back to the first workspace if available
   const folders = vscode.workspace.workspaceFolders;
   if (folders?.length) {
-    return fixDriveCasingInWindows(folders[0].uri.fsPath);
+    // Only file uris are supported
+    let folder = folders.find((folder) => isSafeUriSchemeFunc(folder.uri));
```
Comment (felipecrs) [inline src/utils/path.ts:46]:
```suggestion
    const folder = folders.find((folder) => isSafeUriSchemeFunc(folder.uri));
```

No?

**Review by felipecrs — CHANGES_REQUESTED** [review]
I think we can also rename the PR to something like:

```
fix: not working together with the SSH FS extension
```

WDYT?

**Comment by timonwong** [comment]
> I think we can also rename the PR to something like:
> 
> ```
> fix: not working together with the SSH FS extension
> ```
> 
> WDYT?

I just renamed this PR title, I think after squash the title will be used instead of the original commit messages.

**Pull request merged**